feat(solid): add support for prepareRequestBody in solidjs

Jaikant opened the pull request:
This PR adds support for the experimental_prepareRequestBody option in SolidJS's useChat implementation, aligning it with the existing React functionality. Includes tests & documentation updates

Jaikant commented:
Fixes #4588 for solidjs

lgrammel commented:
Getting a bunch of warnings when running the tests:

```
stderr | src/use-chat.ui.test.tsx > data protocol stream > loading state > should reset loading state on error
computations created outside a `createRoot` or `render` will never be disposed
computations created outside a `createRoot` or `render` will never be disposed

stderr | src/use-chat.ui.test.tsx > data protocol stream > should invoke onFinish when the stream finishes
computations created outside a `createRoot` or `render` will never be disposed
computations created outside a `createRoot` or `render` will never be disposed

stderr | src/use-chat.ui.test.tsx > data protocol stream > id > should clear out messages when the id changes
computations created outside a `createRoot` or `render` will never be disposed
computations created outside a `createRoot` or `render` will never be disposed

stderr | src/use-chat.ui.test.tsx > text stream > should show streamed response
computations created outside a `createRoot` or `render` will never be disposed
computations created outside a `createRoot` or `render` will never be disposed

stderr | src/use-chat.ui.test.tsx > text stream > should invoke onFinish when the stream finishes
computations created outside a `createRoot` or `render` will never be disposed
computations created outside a `createRoot` or `render` will never be disposed

stderr | src/use-chat.ui.test.tsx > onToolCall > should invoke onToolCall when a tool call is received from the server's response
computations created outside a `createRoot` or `render` will never be disposed

stderr | src/use-chat.ui.test.tsx > maxSteps > two steps with automatic tool call > should automatically call api when tool call gets executed via onToolCall
computations created outside a `createRoot` or `render` will never be disposed
computations created outside a `createRoot` or `render` will never be disposed
computations created outside a `createRoot` or `render` will never be disposed
computations created outside a `createRoot` or `render` will never be disposed

stderr | src/use-chat.ui.test.tsx > maxSteps > two steps with error response > should automatically call api when tool call gets executed via onToolCall
computations created outside a `createRoot` or `render` will never be disposed
computations created outside a `createRoot` or `render` will never be disposed
computations created outside a `createRoot` or `render` will never be disposed
computations created outside a `createRoot` or `render` will never be disposed

 ✓ src/use-assistant.ui.test.tsx (7)
 ✓ src/use-chat.ui.test.tsx (22) 360ms
 ✓ src/use-completion.ui.test.tsx (4)
 ✓ src/use-object.ui.test.tsx (9)

```

Jaikant commented:
Okay, double checking.

Jaikant commented:
@lgrammel the issue was because reactive computations were being created outside of a reactive root. The changes I had introduced caused the problem. I have fixed that.  I noticed there are more such problems in `use-assistant` and there are few more warnings coming from tests in `use-assistant.ui.test.tsx`

To debug this issue, I used the following helper code:
```
import { vi } from 'vitest';
import * as Solid from 'solid-js';

/**
 * Helper to wrap a Solid function and warn if called with no current Owner.
 */
function patchFn<T extends Function>(fnName: string, originalFn: T): T {
  return ((...args: any[]) => {
    // getOwner returns the current reactive owner if we are in a component/root
    if (!Solid.getOwner?.()) {
      console.warn(`${fnName} called outside a root!`);
      // Show a stack trace so you know **which code** triggered it:
      console.trace();
    }
    return (originalFn as any)(...args);
  }) as unknown as T;
}

/**
 * We mock 'solid-js' so that all imports of 'solid-js' from anywhere
 * in our code will use these patched versions of the functions.
 */
vi.mock('solid-js', async () => {
  // Grab the real 'solid-js' exports to clone them
  const actual = await vi.importActual<any>('solid-js');
  return {
    ...actual,
    // Patch whichever reactivity functions you want to trace:
    createSignal: patchFn('createSignal', actual.createSignal),
    createMemo:   patchFn('createMemo', actual.createMemo),
    createEffect: patchFn('createEffect', actual.createEffect),
    // etc.
  };
});

```

and I added it to: packages/solid/vitest.config.js
```
   server: { port: 3000 },
   build: { target: 'esnext' },
   test: {
+    setupFiles: ['src/setup.ts'],
     environment: 'jsdom',
     globals: true,
     include: ['src/**/*.ui.test.ts', 'src/**/*.ui.test.tsx'],

```

then I could see the traces of the issue, for example I still can see warnings from use-assistant:
```
stderr | src/use-assistant.ui.test.tsx > stream data stream > should show streamed response
@ai-sdk/solid:test: createSignal called outside a root!
@ai-sdk/solid:test: Trace:
@ai-sdk/solid:test:     at Proxy.<anonymous> (/Users/jai/work/new/ai/packages/solid/src/setup.ts:13:15)
@ai-sdk/solid:test:     at getNode (/Users/jai/work/new/ai/node_modules/.pnpm/solid-js@1.8.7/node_modules/solid-js/store/dist/dev.js:79:42)
@ai-sdk/solid:test:     at setProperty (/Users/jai/work/new/ai/node_modules/.pnpm/solid-js@1.8.7/node_modules/solid-js/store/dist/dev.js:163:15)
@ai-sdk/solid:test:     at updateArray (/Users/jai/work/new/ai/node_modules/.pnpm/solid-js@1.8.7/node_modules/solid-js/store/dist/dev.js:186:33)
@ai-sdk/solid:test:     at /Users/jai/work/new/ai/node_modules/.pnpm/solid-js@1.8.7/node_modules/solid-js/store/dist/dev.js:247:11
@ai-sdk/solid:test:     at runUpdates (/Users/jai/work/new/ai/node_modules/.pnpm/solid-js@1.8.7/node_modules/solid-js/dist/dev.js:893:17)
@ai-sdk/solid:test:     at Proxy.batch (/Users/jai/work/new/ai/node_modules/.pnpm/solid-js@1.8.7/node_modules/solid-js/dist/dev.js:497:10)
@ai-sdk/solid:test:     at setStore (/Users/jai/work/new/ai/node_modules/.pnpm/solid-js@1.8.7/node_modules/solid-js/store/dist/dev.js:245:27)
@ai-sdk/solid:test:     at append (/Users/jai/work/new/ai/packages/solid/src/use-assistant.ts:138:5)
@ai-sdk/solid:test:     at HTMLButtonElement.TestComponent._el$3.$$click (/Users/jai/work/new/ai/packages/solid/src/use-assistant.ui.test.tsx:36:13)

```

I was not sure of the use-assistant code, so I did not risk trying to fix.
